grails-app/controllers/org/pih/warehouse/api/StockMovementItemApiController.groovy

```diff
@@ -29,18 +31,23 @@ class StockMovementItemApiController {
         render([data: stockMovement.lineItems] as JSON)
     }
 
-    def read() {
-        def stockMovementItem = stockMovementService.getStockMovementItem(params.id, params.stepNumber)
+    def read(StockMovementItemParamsCommand command) {
+        def stockMovementItem = stockMovementService.getStockMovementItem(command)
         render([data: stockMovementItem] as JSON)
     }
 
-    def details() {
-        def stockMovementItem = stockMovementService.getStockMovementItem(params.id, params.stepNumber, true)
+    /**
+     * @deprecated As of release 9.2, replaced by {@link #read()} with showDetails query parameter
```

Comment:
I think we should discuss this a bit before deprecating as these are different resources.

This one is the "parent" resource that provides just enough information.
```
/api/stockMovementItems/:id
```
Whereas the other is a nested resource that provides more detailed information. 
```
/api/stockMovementItems/:id/details
```
I would prefer to keep the nested resource, rather than require the client to know to provide a request parameter. But you might be right to do it this way. 

Regardless, we need to come up with an actual convention here. 
https://pihemr.atlassian.net/browse/OBPIH-2649